codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2StreamChannel.java

```diff
@@ -1175,6 +1209,32 @@ public ChannelConfig setRecvByteBufAllocator(RecvByteBufAllocator allocator) {
             super.setRecvByteBufAllocator(allocator);
             return this;
         }
+
+        @Override
+        public Map<ChannelOption<?>, Object> getOptions() {
+            return getOptions(
+                    super.getOptions(),
+                    Http2StreamChannelOption.AUTO_WRITE_WINDOW_UPDATE_FRAME);
+        }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public <T> T getOption(ChannelOption<T> option) {
+            if (option == Http2StreamChannelOption.AUTO_WRITE_WINDOW_UPDATE_FRAME) {
+                return (T) Boolean.valueOf(autoFlowControl);
```

Comment:
Consider keeping names of the option and class variable similar

Reply:
FWIW I am not sure I love the option name... So I am open for suggestions

Reply:
`AUTO_` prefix is common in Netty, then we can highlight that this is related to stream flow control rather than connection flow-control: `AUTO_STREAM_FLOW_CONTROL` or `AUTO_FLOW_CONTROL_FOR_STREAMS`.
Then javadoc for the new option can link to https://datatracker.ietf.org/doc/html/rfc9113#section-5.2 so that users can understand its relationships to `WINDOW_UPDATE` frames.

Reply:
Let's use `AUTO_STREAM_FLOW_CONTROL` that said it might also affect the connection window at the end.